[Yeo Tee Chin] ip

Bluntsord opened the pull request:
#DukePro

> DukePro is the best dukebot of all bots 
> - Myself

  * text-based
  * easy to learn
  * ~~FAST~~ SUPER FAST to use
  ***

All you need to do is,
   1. download it from [here](https://github.com/Bluntsord/ip/releases/new)
   2. double-click it.
   3. add your tasks.
   4. let it manage your tasks for you 😉

And it is **FREE!**

- [x] (Managing tasks) 
- [x] (Managing deadlines (coming soon))

***

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Bluntsord committed:
Created flow diagram for the general guideline of designing the bot and updated READMe

Bluntsord committed:
Merge pull request #1 from Bluntsord/Bluntsord/Docs

Created flow diagram for the general guideline

Bluntsord committed:
Level 1 of dukeBot

Bluntsord committed:
Level 2 of Duke Bot

Bluntsord committed:
Did up java docs for the level 2 bot

Bluntsord committed:
Finished Level 3 of DukeBot

Bluntsord committed:
Completed Level 4 of DukeBot

Bluntsord committed:
adjusted the deadline and event class to have the deadline and time and date

Bluntsord committed:
Completed Testing

Bluntsord committed:
Exceptions were handled from the start, no need to change

Bluntsord committed:
Finished level 6 of dukeBot

Bluntsord committed:
Finished java docs

Bluntsord committed:
Updated Level 6 of Dukebot

Bluntsord committed:
Completed UI Testing 2nd round

Bluntsord committed:
Add Persistent Storage

Bluntsord committed:
Finish level 7. No javadocs

Bluntsord committed:
Add javaDocs

Bluntsord committed:
Add more interactions

Bluntsord committed:
Add some datetime

Bluntsord committed:
Change deadline and event to actual dateTime

Bluntsord committed:
Merge branch 'branch-Level-7'
Add persistent storage

Persistent storage in the form of a text file is added. This is still
part of DataHandlerLayer. Storage will be read when dukebot is booted.

Bluntsord committed:
Merge branch 'branch-Level-8'

Bluntsord committed:
Reformat all codestyles to guidelines

Bluntsord committed:
Add everything to package named duke

Bluntsord committed:
Fix some oop stuff. Much of it has been done already

Bluntsord committed:
Attempt JUnit tests

Bluntsord committed:
Add dumnmy JUnit test

Bluntsord committed:
Add Junit test for Todo and Deadline

Bluntsord committed:
Add tags

Bluntsord committed:
Add javaDocs

Bluntsord committed:
Add all javaDocs

Bluntsord committed:
Follow coding standards

Bluntsord committed:
Add git ignore

Bluntsord committed:
Merge A-Coding-Standard into main

Bluntsord committed:
Miss one javaDoc

Bluntsord committed:
Finish level-9. Add find function

Bluntsord committed:
Merge Level-9 with main

lwlshawn reviewed (commented):
Code looks great! I thought your documentation was really detailed and thorough.

yin72257 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,75 @@
+ */
+public class Command {
+    private Task.TaskType taskType;
+    private ArrayList<String> listOfCommandInputs;
```
Perhaps just name this commandInputs for the plurality.

yin72257 commented on src/main/java/duke/DataHandlerLayer.java:
```diff
@@ -0,0 +1,144 @@
+    /**
+     * In memory storage for log, for history, refer to the history variable
+     */
+    private static ArrayList<Task> log = new ArrayList<>();
```
Shouldn't collection names be plural?

yin72257 commented on src/main/java/duke/DataHandlerLayer.java:
```diff
@@ -0,0 +1,144 @@
+        if (position == 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        ;
```
Isn't this semicolon unnecessary?

yin72257 commented on src/main/java/duke/PersistentStorageHandler.java:
```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.io.*;
```
Shouldn't you import the specific library you used rather than the * sign.

NicolasCwy replied:
I used[ this tutorial](https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcards-for-class) to help me configure Intellij to import my classes easier without converting them into wildcards!

yin72257 reviewed (commented):
There isn't too much coding standard violations. I just found some small nits you might consider changing.

Bluntsord committed:
Merge branch 'master' into add-gradle-support

Merge add-gradle-support with master. This is for using gradle

Bluntsord committed:
Add gradle support and update checkstyle during gradle checkstyle test

NicolasCwy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+        while (true) {
+            String command = UI.scan();
+            try {
+                Logic.checkIfSpecialComand(command);
```
Perhaps you could change this to checkIfValidComand, so that it is more descriptive to what it's doing?

NicolasCwy commented on src/main/java/duke/Logic.java:
```diff
@@ -0,0 +1,134 @@
+        String loggedCommand = packagedCommand.getLog();
+
+        switch (packagedCommand.getTaskType()) {
+        case TODO:
```
Good switch indentation here! Can be easy to miss💯

NicolasCwy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,132 @@
+     *                and packaged into duke.Command. See duke.Command for more.
+     * @return A packaged command for ease of access of infomation
+     */
+    public static Command parse(String command) throws InvalidCommandException {
```
Could consider renaming command to userInput, since you will perform checks later to see if it is in fact a command

NicolasCwy reviewed (commented):
Great work on your first PR!🎉 I liked that you included a diagram of the structure of your program. Overall everything code quality good though there are a few nits with variable names and stray semi-colons!

Bluntsord committed:
Add Basic GUI with some assets

Bluntsord committed:
Add GUI improvements

Bluntsord committed:
Add functioning GUI

Bluntsord committed:
Add some javadocs

Bluntsord committed:
Add some javadocs and some cleaning

Bluntsord committed:
Add javadocs

Bluntsord committed:
Merge branch 'branch-Level-10'
Merge branch-Level-10 into master. Add functioning GUI and images

lwlshawn commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+import javafx.stage.Stage;
+;
+
+/**
+ * Main file for the duke bot. The duke bot keeps track of all your task and allows you to mark as complete.
+ * Call list to see the list of task. Each task can be completed. When a task is completed, it will still appear
+ * when you call list this time, but will be deleted from history for storage purposes the next time it is ran.
+ * Use dukebot to mark down tasks today!
+ */
```
Same comment as before! I thought throughout your documentation was great. Sorry for reposting but 2103 required review comments

lwlshawn reviewed (commented):
for 2103 script

Bluntsord committed:
Try adding jar file

Bluntsord committed:
Handle some bugs found during assertion adding

Bluntsord committed:
Fix more bugs

Bluntsord committed:
Craete general log function

Bluntsord committed:
Unify filter log to accept functions and clean up command class

Bluntsord committed:
Add Jar file creation capabilities

Bluntsord committed:
Apply Slap rules to many functions and debugged event and deadline

Bluntsord committed:
Clean up unused variables

Bluntsord committed:
Remove test function for parser

Bluntsord committed:
Add assertions to parser

Bluntsord committed:
Add assertions to persistent storage

Bluntsord committed:
Merge pull request #3 from Bluntsord/branch-A-Assertions

Merge Branch-a-assertions

Bluntsord committed:
Merge master with branch-A-CodingQuality

Bluntsord committed:
Merge pull request #2 from Bluntsord/branch-A-CodingQuality

Branch a coding quality

Bluntsord committed:
Fix some merging bugs

Bluntsord committed:
Add some help functionality

Bluntsord committed:
Add full help functionality.

Users not able to get help on how to
use the commands or the entire dukeBot itself.

Due to this, there is alot of confusion regarding how to operate
the duke bot

Help Command is added to assist the users and give them a brief overview
commands that the current dukebot supports. Also, when the dukebot first starts,
a reference to the help command appears.

Bluntsord committed:
Add full help functionality

Bluntsord committed:
Merge branch 'master' of https://github.com/Bluntsord/ip

Bluntsord committed:
Add Ui.png

Bluntsord committed:
Set theme jekyll-theme-slate

Bluntsord committed:
Set theme jekyll-theme-slate

Bluntsord committed:
Set theme jekyll-theme-slate

Bluntsord committed:
Set theme jekyll-theme-slate

Bluntsord committed:
Replace Ui.png

Bluntsord committed:
Merge branch 'master' of https://github.com/Bluntsord/ip

Bluntsord committed:
Update README.md

Bluntsord committed:
Remove some dev guide features from README.md

Bluntsord committed:
Add new lines to README.md

Bluntsord committed:
Fix README syntax

Bluntsord committed:
Fix more README syntax

Bluntsord committed:
Clean up checkstyle

Bluntsord committed:
Update and check tests are functional

Bluntsord committed:
Clean up checkstyle for test

Bluntsord committed:
Update Ui.png for mockup

Bluntsord committed:
Update readme with new UI.png

Bluntsord committed:
Update mockup for readme

Bluntsord committed:
Rename to Ui.png
